packages/trader/src/AppV2/Hooks/useOrderDetails.ts

```diff
@@ -0,0 +1,163 @@
+            ? `${data.limit_order.take_profit.order_amount.toFixed(2)} ${data.currency}`
+            : localize('Not set'),
+    };
+};
+
+// For Digits
+const transformMatcherData = (data: TContractInfo) => {
+    const commonFields = getCommonFields(data);
+    return {
+        [localize('Reference ID')]: commonFields[localize('Reference ID')],
+        [localize('Duration')]: `${getDurationTime(data) ?? ''} ${getDurationUnitText(getDurationPeriod(data)) ?? ''}`,
+        [localize('Target')]: getBarrierValue(data),
+        [localize('Stake')]: commonFields[localize('Stake')],
+    };
+};
+// For Accumulators
+const transformAccumulatorData = (data: TContractInfo) => {
+    const commonFields = getCommonFields(data);
+    return {
+        [localize('Reference ID')]: commonFields[localize('Reference ID')],
+        ...{
+            ...((data.is_expired || data.is_sold) && {
+                [localize('Duration')]: commonFields[localize('Duration')],
+            }),
+        },
+        [localize('Growth rate')]: data.growth_rate ? `${getGrowthRatePercentage(data.growth_rate)}%` : '',
+        [localize('Stake')]: commonFields[localize('Stake')],
+        ...{
+            ...(data.limit_order?.take_profit && {
+                [localize('Take Profit')]: data.limit_order?.take_profit?.order_amount
+                    ? `${data.limit_order.take_profit.order_amount} ${data.currency}`
+                    : localize('Not set'),
+            }),
+        },
+    };
+};
+
+// For Vanillas
+const transformVanillaData = (data: TContractInfo) => {
+    const commonFields = getCommonFields(data);
+    return {
+        [localize('Reference ID')]: commonFields[localize('Reference ID')],
+        [localize('Strike Price')]:
+            (isResetContract(data.contract_type) ? addComma(data.entry_spot_display_value) : getBarrierValue(data)) ||
+            ' - ',
+        [localize('Duration')]: `${getDurationTime(data) ?? ''} ${getDurationUnitText(getDurationPeriod(data)) ?? ''}`,
+        [localize('Payout per point')]: commonFields[localize('Payout per point')],
+        [localize('Stake')]: commonFields[localize('Stake')],
+    };
+};
+
+// Map of contract types to their respective transform functions
+const transformFunctionMap: Record<string, (data: TContractInfo) => Record<string, any>> = {
+    [CONTRACT_TYPES.TURBOS.LONG]: transformTurbosData,
+    [CONTRACT_TYPES.TURBOS.SHORT]: transformTurbosData,
+    [CONTRACT_TYPES.MULTIPLIER.DOWN]: transformMultiplierData,
+    [CONTRACT_TYPES.MULTIPLIER.UP]: transformMultiplierData,
+    [CONTRACT_TYPES.MATCH_DIFF.MATCH]: transformMatcherData,
+    [CONTRACT_TYPES.MATCH_DIFF.DIFF]: transformMatcherData,
+    [CONTRACT_TYPES.EVEN_ODD.EVEN]: transformMatcherData,
+    [CONTRACT_TYPES.EVEN_ODD.ODD]: transformMatcherData,
+    [CONTRACT_TYPES.OVER_UNDER.OVER]: transformMatcherData,
+    [CONTRACT_TYPES.OVER_UNDER.UNDER]: transformMatcherData,
+    [CONTRACT_TYPES.RESET.CALL]: transformRiseData,
```

Comment:
`CONTRACT_TYPES.RESET.CALL` is a contract from SmartTrader. Are you going to handle SmartTrader contracts right now? If yes, than we need to add more field to this object (at least for `CONTRACT_TYPES.RESET.PUT`).

Reply:
I don't think there's design ready for these contracts, but we can keep this one if it works